tests/testthat/test-helper_functions.R

```diff
@@ -15,55 +13,53 @@ data(X.class)
 data(Y.class)
 
 test_that("Recycling generates a warning", {
-  expect_error(caretEnsemble:::wtd.sd(matrix(1:10, ncol=2), w=1))
+  expect_error(caretEnsemble:::wtd.sd(matrix(1:10, ncol = 2), w = 1))
 })
 
 test_that("No predictions generates an error", {
-  skip_if_not_installed("randomForest")
-  skip_if_not_installed("gbm")
-  skip_if_not_installed("plyr")
-  skip_if_not_installed("glmnet")
   suppressWarnings(
     models_multi <- caretList(
       iris[, 1:2], iris[, 5],
-      tuneLength=1, verbose=FALSE,
-      methodList=c("rf", "gbm"),
-      trControl=trainControl(method="cv", number=2, savePredictions="final", classProbs=TRUE))
+      tuneLength = 1, verbose = FALSE,
+      methodList = c("rf", "gbm"),
+      trControl = trainControl(method = "cv", number = 2, savePredictions = "final", classProbs = TRUE)
+    )
   )
   expect_error(check_caretList_model_types(models_multi))
 
   suppressWarnings(
     models <- caretList(
-      iris[, 1:2], factor(ifelse(iris[, 5]=="setosa", "Yes", "No")),
-      tuneLength=1, verbose=FALSE,
-      methodList=c("rf", "gbm"),
-      trControl=trainControl(method="cv", number=2, savePredictions="final", classProbs=TRUE))
+      iris[, 1:2], factor(ifelse(iris[, 5] == "setosa", "Yes", "No")),
+      tuneLength = 1, verbose = FALSE,
+      methodList = c("rf", "gbm"),
+      trControl = trainControl(method = "cv", number = 2, savePredictions = "final", classProbs = TRUE)
+    )
   )
   new_model <- train(
-    iris[, 1:2], factor(ifelse(iris[, 5]=="setosa", "Yes", "No")),
-    tuneLength=1,
-    method=c("glmnet"),
-    trControl=trainControl(method="cv", number=2, savePredictions="none", classProbs=TRUE)
+    iris[, 1:2], factor(ifelse(iris[, 5] == "setosa", "Yes", "No")),
+    tuneLength = 1,
+    method = c("glmnet"),
+    trControl = trainControl(method = "cv", number = 2, savePredictions = "none", classProbs = TRUE)
   )
   models2 <- c(new_model, models)
   models3 <- c(models, new_model)
   check_caretList_model_types(models)
   expect_error(check_caretList_model_types(models2))
-  #expect_error(check_caretList_model_types(models3)) #THIS IS A BUG THAT NEEDS FIXING!!!!!!!!!!
```

Comment:
The fix above that checks all models, not just the first in the list, fixes this bug

Reply:
https://github.com/zachmayer/caretEnsemble/pull/273/files?diff=unified&w=0#diff-cf18b2f78154f3fdc2849178852c8f5b3fbfcffb589562f15d16d55edea57a2eL101